DataFlashBlockDevice.cpp

```diff
@@ -127,6 +128,8 @@ enum dummy {
     DATAFLASH_HIGHEST_FREQUENCY_BYTES = 2
 };
 
+SingletonPtr<PlatformMutex> DataFlashBlockDevice::_mutex;
```

Comment:
Why singleton pointer? To protect the SPI bus? Should you use the actual SPI locking?

Reply:
Ah just saw the above comment. Shouldn't you then use a member mutex combined with the SPI locking? Unfortunately this won't synchronize with another device on the same SPI bus

Reply:
Ah, so you are saying every time we lock the dataflash mutex, we also lock the SPI mutex for the entire duration?

Reply:
Well, you only need to lock the SPI mutex for as long as you need the bus uninterrupted. But if there's additional state you need to protect outside of the SPI bus you can use an additional mutex that wraps the logic and bus.

If the logic modification quick, you could just use the SPI bus's lock for all synchronization, but that's up to you.